examples/usercmodule/cexample/micropython.cmake

```diff
@@ -0,0 +1,21 @@
+# Create an INTERFACE library for our C module.
+add_library(usermod_cexample INTERFACE)
+
+# Add our source files to the lib
+target_sources(usermod_cexample INTERFACE
+    ${CMAKE_CURRENT_LIST_DIR}/examplemodule.c
+)
+
+# Add the current directory as an include directory.
+target_include_directories(usermod_cexample INTERFACE
+    ${CMAKE_CURRENT_LIST_DIR}
+)
+
+# Enable the module automatically by adding the relevant compile definitions.
+target_compile_definitions(usermod_cexample INTERFACE
+    MODULE_CEXAMPLE_ENABLED=1
+)
```

Comment:
What was the reasoning to enable the module automatically?  The makefile (.mk) way doesn't do this, so this will be an inconsistency.  Also, how would one disable (override) this value?

Reply:
I'm going to point to ulab here as probably the most high profile example to back up my reasoning:

https://github.com/v923z/micropython-ulab/blob/5890f0e193ef91ba584f114afc4fe3a9e8de6a99/code/micropython.mk#L33

But there's also:

https://github.com/devbis/st7789_mpy/blob/master/st7789/micropython.mk
https://github.com/mocleiri/tensorflow-micropython-examples/blob/5a8656b658017538166bfdd92c89f58124014a75/micropython-modules/audio_frontend/micropython.mk
https://github.com/glennrub/ubluepy/blob/3a1ae04c98a2a460aab84ba13294825bdd43596b/micropython.mk
https://github.com/mcauser/micropython-pcd8544-c/blob/5ee83f96b83236c81f11c35cd43fccf106bede11/micropython.mk

(though I bet you can find just as many examples of it *not* being included as I can of it being included, so I have to venture that Tensorflow and ulab are pretty strong candidates for setting precedent)

It's clear I'm not alone in thinking the source of truth for enabling modules should be the list of modules that are supplied to the build- since it's a user-supplied build argument. Requiring the user to enable modules in two places is a potential trap for beginners (one that, yes, I fell into when first learning to build user modules against MicroPython). I'd be inclined to add something like the line from ulab to the `micropython.mk` files for user modules to establish this as best practise.

Is there likely to be a case where a user includes a module but *doesn't* want it enabled? I guess our very liberal collection of modules might be one example. We have a main `.cmake` file that includes all of our modules: https://github.com/pimoroni/pimoroni-pico/blob/main/micropython/modules/micropython.cmake

And then we build our version of the port *just* by including this list, without needing to apply any pre-build patches to `mpconfigport.h` or maintain our own fork with a different list of enabled modules.

As an end-user if you didn't want *all the things* from our top-level `micropython.cmake` you would have to do something like `#undef MODULE_NAME_ENABLED` in `mpconfigport.h` to explicitly disable modules on a case by case basis, but the core premise of my argument is that someone building a custom MicroPython with their choice of modules:

1. Shouldn't have to edit or even care about `mpconfigport.h`
2. Shouldn't be surprised when they use only the `USER_C_MODULES` argument, but the  modules are missing from the build

I'm not particularly concerned one way or another regarding your in-house examples, since whether or not they enable automatically doesn't change things *for me* but I'm willing to make the case that the original design intent, and how the wider community have embraced it have diverged somewhat and perhaps (at the risk of making more work for myself) the example `micropython.mk` files and documentation should update to reflect this.